### PR TITLE
Update to Conda 23.7.2 & conda-libmamba-solver 23.7.0

### PR DESCRIPTION
**Pull request by jakirkham**
Checklist
* [ ] Used a [personal fork of the feedstock to propose changes](https://conda-forge.org/docs/maintainer/updating_pkgs.html#forking-and-pull-requests)
* [ ] Bumped the build number (if the version is unchanged)
* [ ] Reset the build number to `0` (if the version changed)
* [ ] [Re-rendered]( https://conda-forge.org/docs/maintainer/updating_pkgs.html#rerendering-feedstocks ) with the latest `conda-smithy` (Use the phrase <code>@<space/>conda-forge-admin, please rerender</code> in a comment in this PR for automated rerendering)
* [ ] Ensured the license file is being packaged.

### CODE REVIEW

**Comment by jakirkham** [comment]
cc @jaimergp @hmaarrfk

**Comment by jaimergp** [comment]
Unfortunately we need to wait for a hypothetical mamba 1.4.10 to include https://github.com/mamba-org/mamba/pull/2732, which fixes #475.

**Comment by h-vetinari** [comment]
> Unfortunately we need to wait for a hypothetical mamba 1.4.10 to include [mamba-org/mamba#2732](https://github.com/mamba-org/mamba/pull/2732), which fixes #475.

Would also fix https://github.com/conda-forge/boa-feedstock/issues/75, which is necessary to unblock some fixes in conda-build 3.26.

**Comment by jaimergp** [comment]
It looks like it will be cut soon (today?): https://github.com/mamba-org/mamba/issues/2766

**Comment by h-vetinari** [comment]
mamba 1.5.0 is live in conda-forge

**Comment by jaimergp** [comment]
mamba 1.5.0 available. Let's rerun.

**Comment by h-vetinari** [comment]
That's twice I snuck in _juuust_ before you today (in my timezone) 😝

**Comment by h-vetinari** [comment]
I think this is somehow running into this constraint on mamba <1.4.4:
https://github.com/conda-forge/mamba-feedstock/blame/67d9230a2570dd951c84460da7fe2f0974bc3f6e/recipe/meta.yaml#L125

File: Miniforge3/construct.yaml [hunk 1]
```diff
@@ -1,6 +1,6 @@
 {% set name = os.environ.get("MINIFORGE_NAME", "Miniforge3") %}
-{% set version = os.environ.get("MINIFORGE_VERSION", "23.3.1-0") %}
-{% set conda_libmamba_solver_version = "23.3.0"%}
+{% set version = os.environ.get("MINIFORGE_VERSION", "23.7.2-0") %}
+{% set conda_libmamba_solver_version = "23.7.0"%}
 {% set mamba_version = "1.4.2"%}
```
Comment (jaimergp) [inline Miniforge3/construct.yaml:4]:
```suggestion
{% set mamba_version = "1.5.0"%}
```

**Comment by h-vetinari** [comment]
Ah, and this conflicts with https://github.com/conda-forge/conda-libmamba-solver-feedstock/blob/d6ef0c91d0ecae6d01e84963b79e866de604c592/recipe/meta.yaml#L28-L29, because it's impossible to satisfy both constraints there. Satisfying `conda >=23.5` implies `libmamba >=1.4.4`, and vice versa `libmamba >=1.4.1,<1.4.4` (there's no build for 1.4.3) implies `conda <23.4`.

**Comment by h-vetinari** [comment]
Ah, and now you're back to https://github.com/conda-forge/boa-feedstock/issues/75, because boa caps to `mamba <1.4.2`... 😑

**Comment by jakirkham** [comment]
Superseded by PR: https://github.com/conda-forge/miniforge/pull/523

**Pull request closed without merging**